pkg/controllers/cronfederatedhpa/cronfederatedhpa_job.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2023 The Karmada Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cronfederatedhpa
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gorhill/cronexpr"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	autoscalingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/autoscaling/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+type CronFederatedHPAJob struct {
+	client        client.Client
+	eventRecorder record.EventRecorder
+
+	namespaceName types.NamespacedName
+	rule          autoscalingv1alpha1.CronFederatedHPARule
+}
+
+func NewCronFederatedHPAJob(client client.Client, eventRecorder record.EventRecorder,
+	namespaceName types.NamespacedName, rule autoscalingv1alpha1.CronFederatedHPARule) *CronFederatedHPAJob {
+	return &CronFederatedHPAJob{
+		client:        client,
+		eventRecorder: eventRecorder,
+		namespaceName: namespaceName,
+		rule:          rule,
+	}
+}
+
+func (c *CronFederatedHPAJob) Run() {
+	klog.V(4).Infof("Start to handle CronFederatedHPA %s", c.namespaceName)
+	defer klog.V(4).Infof("End to handle CronFederatedHPA %s", c.namespaceName)
+
+	var err error
+	cronFHPA := &autoscalingv1alpha1.CronFederatedHPA{}
+	err = c.client.Get(context.TODO(), c.namespaceName, cronFHPA)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("CronFederatedHPA %s not found", c.namespaceName)
+		} else {
+			klog.Errorf("Get CronFederatedHPA failed: %v", err)
```

Comment:
I think this `err` should also be recorded in the `FailedExecutionHistory`.

Reply:
We can't get this object, So I think we can't record it.

Reply:
I think we should record it, but if we can't get the object, how can we add the failed history record? Do you have any idea?

Reply:
TODO added